lib/fluent/event.rb

```diff
@@ -162,5 +182,21 @@ def to_msgpack_stream
       @data
     end
   end
-end
 
+  module ChunkMessagePackEventStreamer
+    include MessagePackFactory::Mixin
+    # chunk.extend(ChunkEventStreamer)
+    #  => chunk.each{|time, record| ... }
+    def each(&block)
```

Comment:
Original implementaion resuces IOError from MessagePack::Unpacker.
Is removing `rescue` safe?

```rb
def msgpack_each(&block)
  open do |io|
    u = msgpack_factory.unpacker(io)
    begin
      u.each(&block)
    rescue EOFError
    end
  end
end
```